### DISCUSSION TITLE
Mixer-TTS sup_data_path and sup_data_types

### DISCUSSION THREAD

**Discussion by oytunturk** [NVIDIA-NeMo/NeMo#3793] [Q&A] [answered]
Are there any examples/documentation on how to generate supplementary data for Mixer-TTS training? Is 'pitch' data optional or is it extracted automatically during training? How about 'align_prior_matrix'? I guess that must be coming from an ASR model.

Thanks!

**Comment by redoctopus** [comment, marked as answer]
You can use this script to do it for the datasets in the `.../dataset_processing/tts/` directory (or your own if you build your own YAML configs): https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/extract_sup_data.py

But yes, essentially the TTSDataset class will extract these automatically during training, which is what the above script uses to do so. Specifically they happen in these functions in TTSDataset: https://github.com/NVIDIA/NeMo/blob/71c51ec24d0b2774dcdd7dcbbf91c6875ce5383b/nemo/collections/tts/torch/data.py#L330-L370

**Reply by oytunturk** [reply]
Thank you!
